### PR TITLE
Aggregator compression

### PR DESCRIPTION
**Pull request by tsharju**
Enables snappy compression for traffic between m3coordinator and m3aggregator when using m3msg protocol.

### CODE REVIEW

File: src/msg/consumer/handlers_test.go [hunk 1]
```diff
@@ -112,33 +116,35 @@ func TestServerWithConsumeFn(t *testing.T) {
...
+		l, err := net.Listen("tcp", "127.0.0.1:0")
+		require.NoError(t, err)
 
-	// Set a large ack buffer size to make sure the background go routine
-	// can flush it.
-	opts := testOptions().SetAckBufferSize(100)
-	s := server.NewServer("a", NewConsumerHandler(consumeFn, opts), server.NewOptions())
-	require.NoError(t, err)
-	s.Serve(l)
+		// Set a large ack buffer size to make sure the background go routine
```
Comment (fingon) [inline src/msg/consumer/handlers_test.go:123]: tabs? go fmt :-)

File: src/msg/producer/writer/consumer_writer_test.go [hunk 2]
```diff
@@ -54,44 +55,49 @@ var (
...
+		lis, err := net.Listen("tcp", "127.0.0.1:0")
+		require.NoError(t, err)
+		defer lis.Close()
 
-	mockRouter := NewMockackRouter(ctrl)
+		ctrl := xtest.NewController(t)
+		defer ctrl.Finish()
 
-	opts := testOptions()
+		mockRouter := NewMockackRouter(ctrl)
```
Comment (fingon) [inline src/msg/producer/writer/consumer_writer_test.go:71]: tabs here too.

**Review by fingon — CHANGES_REQUESTED** [review]
LGTM except for need for go fmt'ing.

File: src/msg/consumer/handlers_test.go (hunk 1 above)
Comment (tsharju) [reply src/msg/consumer/handlers_test.go:123]: Hmm... I think my go fmt is using tabs. Not sure if that is the default, but if I run it all spaces covert to tabs.

Comment (fingon) [reply src/msg/consumer/handlers_test.go:123]: Right, too much Python recently ;)

**Review by fingon — APPROVED** [review]

**Pull request merged**